File: library_CCs/Geometry.c
#include "Geometry.h"

float calcModule(Point _point)
{
  return sqrt(pow(_point.x, 2) + pow(_point.y, 2));
}

float calcPhase(Point _point)
{
  return atan2(_point.y, _point.x);
}

float fitAngle(float _angle)
{
  if (_angle > 0)
  {
    return _angle % (2 * M_PI);
  }
  else
  {
    return 2 * M_PI - _angle % (2 * M_PI);
  }
}

float micronRound(float _number)
{
  if (_number * 10000000 % 10 > 4)
  {
    return (float)((int)(_number * 1000000)) / 1000000;
  }
  else
  {
    return (float)((int)(_number * 1000000 + 1)) / 1000000;
  }
}

File: provaPointC/provaPointC/provaPointC/Geometry.vb
Module Geometry
    Public Function midPoint(_point1 As PointC, _point2 As PointC) As PointC
        Return New PointC((_point1.getX + _point2.getX) / 2, ((_point1.getY + _point2.getY) / 2))
    End Function

    Public Function pointDistance(_point1 As PointC, _point2 As PointC) As Double
        Return Math.Sqrt(Math.Pow(_point1.getX - _point2.getX, 2) + Math.Pow(_point1.getY - _point2.getY, 2))
    End Function

    Public Function pointTransport(_point As PointC, _length As Double, _phase As Double) As PointC
        If Math.Abs(_phase) = degToRad(90) Then
            Return New PointC(_point.getX, _point.getY + _length * Math.Sin(_phase))
        ElseIf _phase = 0 Or _phase = degToRad(180) Then
            Return New PointC(_point.getX + Math.Cos(_phase) * _length, _point.getY)
        Else
            Return New PointC(_point.getX + Math.Cos(_phase) * _length, _point.getY + _length * Math.Sin(_phase))
        End If
        Return New PointC(_point.getX + _length * Math.Cos(_phase), _point.getY + _length * Math.Sin(_phase))
    End Function

    Public Function degToRad(_angle As Double)
        Return _angle * Math.PI / 180
    End Function

    Public Function radToDeg(_angle As Double)
        Return _angle * 180 / Math.PI
    End Function
End Module
